client/rest-high-level/src/main/java/org/elasticsearch/client/IndicesRequestConverters.java

```diff
@@ -394,6 +395,26 @@ static Request getTemplates(GetIndexTemplatesRequest getIndexTemplatesRequest) t
         return request;
     }
 
+    static Request templatesExist(GetIndexTemplatesRequest getIndexTemplatesRequest) {
+        if (getIndexTemplatesRequest.names() == null || getIndexTemplatesRequest.names().length == 0) {
+            throw new IllegalArgumentException("Must provide at least one index template name");
+        }
+
+        if (Arrays.stream(getIndexTemplatesRequest.names()).anyMatch(indexTemplate -> Strings.hasText(indexTemplate) == false)) {
```

Comment:
Can you create a copy of `GetIndexTemplatesRequest` and move the validation into that instead? Put it in `o.e.client.indices`. Also remove all of the to/from transport action stuff, and do your best to make constructor params final.

Reply:
Will do